Improve SolidusBraintree README

gsmendoza opened the pull request:
## Summary

Closes https://github.com/solidusio/solidus_braintree/issues/123.

## Goals

* Provide overview of what README covers.
* Improve navigability of README.
* Tie SolidusBraintree payment options to corresponding Braintree documentation.
* Add section for Credit card payment and Hosted Fields solution.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

gsmendoza commented:
@kennyadsl Can you review? If you have other ideas on how we can improve the README, please let me know. Thanks!

waiting-for-dev commented on README.md:
```diff
@@ -3,10 +3,25 @@
 
 🚧 This extension is currently only compatible with the `solidus_starter_frontend` 🚧
 
+## Table of Contents
```
I'm not a fan of having a manually maintained TOC, as GitHub automatically generates that:

<img src="https://user-images.githubusercontent.com/52650/224228314-97ae8552-e1a5-411c-b1a9-826aed16b0fc.png" width="350" />

However, it's not a feature super well-known, so no strong opinion here :slightly_smiling_face:

gsmendoza replied:
@waiting-for-dev I wasn't aware about that GitHub feature :p Testing out that TOC, I think it's useful for jumping to a section in the README that you already know. However, it's not as useful as a way to get a high level overview of the README, since it's not large enough to view all of the headings. As such, I think it still makes sense to keep the manually generated table of contents. 

@kennyadsl Do you have any thoughts about the TOC?

kennyadsl replied:
I also didn't know about this feature, wow! No strong opinions here either, but I'd like to have some sort of consistency across the extensions' README and, the others do not have a TOC. To be fair, it won't be having or not having this TOC to make the README uniform with the other extensions, we will probably need to create a definitive format and update them in bulk to accomplish this, so it's more a thought for later. That said, up to you @gsmendoza, if you like it we can keep it!

waiting-for-dev reviewed (approved):
Thanks, @gsmendoza. I left a comment to your best judgment :slightly_smiling_face: 

Now that we're going through the extensions, it might be a good time to document the ones officially maintained on the main guides. If we do that, I'd replace the READMEs with just a link to the main page (to avoid maintaining two sources of truth). Thoughts, @kennyadsl, @elia, @rainerdema?

kennyadsl commented on README.md:
```diff
@@ -3,10 +3,25 @@
-`solidus_braintree` is an extension that adds support for using [Braintree](https://www.braintreepayments.com) as a payment source in your [Solidus](https://solidus.io/) store. It supports Apple Pay, PayPal, and credit card transactions.
+`solidus_braintree` is an extension that adds support for using [Braintree](https://www.braintreepayments.com) as a payment source in your [Solidus](https://solidus.io/) store. It supports Apple Pay, PayPal, Venmo, and credit card transactions. The extension uses Braintree's [JavaScript v3 SDK](https://braintree.github.io/braintree-web/current/).
 
 🚧 This extension is currently only compatible with the `solidus_starter_frontend` 🚧
```
Maybe we can remove this line and add a "Compatibility" section, where we explain which extension version is compatible with which Solidus version?

kennyadsl reviewed (approved):
Approved, but left a couple of comments.

gsmendoza commented on README.md:
```diff
@@ -3,9 +3,12 @@
 
-`solidus_braintree` is an extension that adds support for using [Braintree](https://www.braintreepayments.com) as a payment source in your [Solidus](https://solidus.io/) store. It supports Apple Pay, PayPal, and credit card transactions.
+`solidus_braintree` is an extension that adds support for using [Braintree](https://www.braintreepayments.com) as a payment source in your [Solidus](https://solidus.io/) store. It uses Braintree's [JavaScript v3 SDK](https://braintree.github.io/braintree-web/current/) to support the following Braintree payment method types:
 
-🚧 This extension is currently only compatible with the `solidus_starter_frontend` 🚧
+* [Apple Pay](https://developer.paypal.com/braintree/docs/guides/apple-pay/overview)
+* [Venmo](https://developer.paypal.com/braintree/docs/guides/venmo/overview)
+* [PayPal Checkout](https://developer.paypal.com/braintree/docs/guides/paypal/overview/javascript/v3)
+* [Credit Cards](https://developer.paypal.com/braintree/docs/guides/credit-cards/overview) via [Hosted Forms](https://developer.paypal.com/braintree/docs/guides/hosted-fields/overview)
```
I replaced the table of contents with links to the PayPal documentation of the payment method types.

gsmendoza commented on README.md:
```diff
@@ -3,9 +3,12 @@
-`solidus_braintree` is an extension that adds support for using [Braintree](https://www.braintreepayments.com) as a payment source in your [Solidus](https://solidus.io/) store. It supports Apple Pay, PayPal, and credit card transactions.
+`solidus_braintree` is an extension that adds support for using [Braintree](https://www.braintreepayments.com) as a payment source in your [Solidus](https://solidus.io/) store. It uses Braintree's [JavaScript v3 SDK](https://braintree.github.io/braintree-web/current/) to support the following Braintree payment method types:
 
-🚧 This extension is currently only compatible with the `solidus_starter_frontend` 🚧
```
I removed the `solidus_starter_frontend` compatibility warning. It's already mentioned in "Upgrading from SolidusPaypalBraintree 1.2.0 to SolidusBraintree 2.0.0".

gsmendoza commented on README.md:
```diff
@@ -59,10 +64,19 @@ The gem has undergone two major changes: 1) it's been renamed to SolidusBraintre
+| SolidusBraintree | Solidus | Frontend               |
+|------------------|---------|------------------------|
+| 2.0              | 3.4     | SolidusStarterFrontend |
+| 1.3              | 3.4     | SolidusFrontend        |
```
SolidusBraintree 1.3 hasn't been released yet. Well need to update the 1.x branch and backport all the master changes up to https://github.com/solidusio/solidus_braintree/pull/99.

gsmendoza replied:
FYI, added https://github.com/solidusio/solidus_braintree/issues/127.

gsmendoza commented:
@kennyadsl @kennyadsl I rebased the PR and made two changes:

1. Replaced the TOC with a list of the payment method types.
2. Added a compatibility section.

Appreciate if you can take one more look! Thanks!

gsmendoza commented on README.md:
```diff
@@ -22,46 +25,19 @@ bundle
+## Upgrading
 
-This gem is the result of merging two gems: the original SolidusBraintree gem, which was deprecated in 2020, and the SolidusPaypalBraintree gem, which was written from scratch to replace it. The updated codebase is based solely on the SolidusPaypalBraintree code. However, we've renamed the gem back to SolidusBraintree to make it clear that it's backed by Braintree. The name also differentiates it ...
+See https://github.com/solidusio/solidus_braintree/wiki/Upgrading.
```
Moved upgrading documentation to wiki.

kennyadsl commented on README.md:
```diff
@@ -22,46 +25,19 @@ bundle
-Considering that SolidusBraintree 2.0.0 is based on the SolidusPaypalBraintree codebase, please be warned that migrating directly from SolidusBraintree 1.2.0 to 2.0.0 would lead to breaking changes. Since SolidusBraintree 1.x was already [deprecated](https://github.com/solidusio/solidus_braintree/tree/v1.x#deprecation-notice-warning-construction), we don't provide at the moment a ready-made solut ...
+| Version                    | Latest Supported Solidus | Frontend         |
+|----------------------------|--------------------------|------------------|
+| SolidusBraintree 2.0       | 3.4                      | SolidusFrontend  |
```
Why Solidus 3.4? I think they are compatible with all versions of Solidus, isn't it?

gsmendoza replied:
@kennyadsl Yes, I updated the title to  "Latest Supported Solidus" to clarify that this is the latest version of Solidus that the extension supports.

That being said, considering that the table shows that there are no differences concerning Solidus dependencies between SolidusBraintree 2.0 and SolidusPaypalBraintree 1.2, perhaps there's no reason to add the table? What do you think? What information can we add that can make the table more useful?
